include/mbgl/storage/resource.hpp

```diff
@@ -20,17 +20,42 @@ class Resource {
         SpriteJSON
     };
 
-    Resource(Kind kind_, const std::string& url_)
+    struct TileData {
+        std::string urlTemplate;
+        float pixelRatio;
```

Comment:
Does this need to be a float? Are we supporting assets that have a fractional pixel ratio?